### DISCUSSION TITLE
How to modify?

### DISCUSSION THREAD

**Discussion by DenBru456** [pdfarranger/pdfarranger#798] [Q&A] [answered]
I need to modify program(visual and some drag and drop parameters also scale) but How will I do this. Well how to modify downloaded open source and after it install it? I can modify vut can't install on windows

**Comment by jeromerobert** [comment, marked as answer]
For Ubuntu follow https://github.com/pdfarranger/pdfarranger#install-from-source and replace the very last step by https://github.com/pdfarranger/pdfarranger#for-developers.

For Windows go to https://github.com/pdfarranger/pdfarranger/blob/main/Win32.md.

**Reply by DenBru456** [reply]
On windows i have problem cause i'm from ukraine and can't have full msys2(internet blocks russian and vpn is not helping) so on windows it's broken for Ukraine users

**Reply by DenBru456** [reply]
Welll in msys2 of course

**Reply by DenBru456** [reply]
The app is working fine. But I cannot launch setup.py(in msys2) it keeps saying cannot find pikepdf but i installed it

**Reply by DenBru456** [reply]
Step by step for windows( with developer method and all depecies)

**Comment by DenBru456** [comment]
Nah vpn was shitty and fake

**Comment by DenBru456** [comment]
Do you know good vpn for windows?

**Comment by DenBru456** [comment]
I have win11

**Comment by DenBru456** [comment]
Is that pikepdf error (another guy already asked)still there or there's a way to succefly install pikepdf on msys2?

**Comment by DenBru456** [comment]
I have the same as that guy

**Reply by jeromerobert** [reply]
What are you talking about ? Which error ? Which guy ?

**Comment by jeromerobert** [comment]
I deleted you other thread about Ubuntu, please keep discussion in that thread. Also this is a forum not a chat, so please write multiple sentences in each posts.

**Reply by jeromerobert** [reply]
@DenBru456 Again keep the discussion in this thread. If you keep opening new threads I ban you from PDF Arranger because you are flooding us.

**Reply by jeromerobert** [reply]
User banned
